frontend/src/ui/wallet/transfer/transfer-tx-builder.js

```diff
@@ -11,11 +11,12 @@ import {estimateFee} from '../../../util/fee-estimator'
  */
 export async function prepareTransferTx(transfer) {
     if (!transfer.hasSufficientBalance) return null
-    const {accountData, address} = accountLedgerData
+    const {accountData} = accountLedgerData
+    const currentFee = (typeof transfer.fee === Number) ? transfer.fee : resolveConfidenceFee(transfer.fee)
 
     const builder = new TransactionBuilder(accountData, {
         networkPassphrase: resolveNetworkParams({network: transfer.network}).network,
-        fee: await estimateFee(transfer.network)
+        fee: currentFee ? currentFee : await estimateFee(transfer.network)
```

Comment:
This condition never works because you always initialize `currentFee`
```suggestion
        fee: currentFee
```

Reply:
If something wrong, resolveConfidenceFee return false and in this case need this condition